app/models/concerns/tags_substitution_concern.rb

```diff
@@ -354,15 +354,18 @@ def parse_tags(text)
   end
 
   def used_tags_and_libelle_for(text)
-    parse_tags(text).filter_map do |token|
-      case token
-      in { tag: tag, id: id }
-        [id, tag]
-      in { tag: tag }
-        [tag]
-      else
-        nil
-      end
+    # MD5 should be enough and it avoids long key
+    Rails.cache.fetch("tags_substitution_concern:#{Digest::MD5.hexdigest(tag)}", expires_in: 1.day) do
```

Comment:
je suis perdu mais j'aurais pensé faire un 

```suggestion
    Rails.cache.fetch("tags_substitution_concern:#{Digest::MD5.hexdigest(text)}", expires_in: 1.day) do
```

Reply:
pareil je pige pas d'où vient `tag`

sinon : 
```ruby
Rails.cache.fetch(Digest::MD5.hexdigest(text), namespace: "tags_substitution_concern", expires_in: 1.day) 

# ou 

Rails.cache.fetch(["tags_substitution_concern", Digest::MD5.hexdigest(text)], expires_in: 1.day) 
```

- le premier peut faciliter un possible futur sharding du cache 
- le second est AMHO  plus lisible et maintenable


Et sinon c'est un détail mais est-ce que ce ne serait pas plus clair d'utiliser comme ns `parse_tags` qui est encore + explicite, plutôt que le nom du concern (dont on pourrait vouloir cacher plusieurs méthodes) ?

Reply:
vous avez raison tous les 2, j'ai complètement craqué avec les `tags`, c'est `text` qu'il fallait utiliser.

et du coup, je prends la remarquable version de Mr Colin :     
`Rails.cache.fetch(["parse_tags", Digest::MD5.hexdigest(text)], expires_in: 1.day)`